### PR TITLE
Convert /api/a-f folder to Markdown (es)

### PR DESCRIPTION
**Pull request by Graywolf9**
### Description

Converting html to md from api/a-e folders

### Motivation

the chore of conversion

### Related issues and pull requests

Relates to https://github.com/mdn/translated-content/issues/8635

### CODE REVIEW

File: files/es/web/api/navigator/getusermedia/index.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ function(localMediaStream) {
 
 ### errorCallback
 
-La función `getUserMedia `llama a la función indicada en el errorCallback con un `código` como argumento. Los códigos de error se describen a continuación:
+La función `getUserMedia`llama a la función indicada en el errorCallback con un `código` como argumento. Los códigos de error se describen a continuación:
```
Comment (nschonni) [inline files/es/web/api/navigator/getusermedia/index.md:82]: Need to watch out for these ones, where Markdownlint trims space inside the code span, but it's missing from the outside

Comment (Graywolf9) [reply files/es/web/api/navigator/getusermedia/index.md:82]: Thank you so much @nschonni did you check these manually or have an script?

Comment (nschonni) [reply files/es/web/api/navigator/getusermedia/index.md:82]: Just manually unfortunately

Comment (Graywolf9) [reply files/es/web/api/navigator/getusermedia/index.md:82]: Thank you so much, I'm going to start reviewing the whole PR now as the past PRs

File: files/es/web/api/analysernode/index.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+---
+title: AnalyserNode
+slug: Web/API/AnalyserNode
+translation_of: Web/API/AnalyserNode
+---
+{{APIRef("Web Audio API")}}
+
+La interfaz **`AnalyserNode`\*\*** \*\*representa un nodo habilitado para proveer frecuencia en tiempo real y analisis de tiempo-dominio. Es un {{domxref("AudioNode")}} que pasa el flujo de audio sin modificación desde el origen de entrada a la salida, pero, te permite obtener los datos generados, procesarlos, y crear visualizaciones de audio.
```
Comment (Graywolf9) [inline files/es/web/api/analysernode/index.md:8]:
```suggestion
La interfaz **`AnalyserNode`** representa un nodo habilitado para proveer frecuencia en tiempo real y analisis de tiempo-dominio. Es un {{domxref("AudioNode")}} que pasa el flujo de audio sin modificación desde el origen de entrada a la salida, pero, te permite obtener los datos generados, procesarlos, y crear visualizaciones de audio.
```

File: files/es/web/api/cachestorage/keys/index.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+title: CacheStorage.keys()
+slug: Web/API/CacheStorage/keys
+translation_of: Web/API/CacheStorage/keys
+---
+{{APIRef ("API de Service Workers")}}
+
+El **`keys`\*\***`()`\*\*método de la interfaz {{domxref ("CacheStorage")}} devuelve un {{jsxref ("Promise")}} que se resolverá con una matriz que contiene las cadenas correspondientes a todos los {{domxref ("Cache")}} objetos rastreados por el objeto {{domxref ("CacheStorage")}} en el orden en que fueron creados. Use este método para iterar sobre una lista de todos los objetos {{domxref ("Cache")}}.
```
Comment (Graywolf9) [inline files/es/web/api/cachestorage/keys/index.md:8]:
```suggestion
El **`keys()`** método de la interfaz {{domxref ("CacheStorage")}} devuelve un {{jsxref ("Promise")}} que se resolverá con una matriz que contiene las cadenas correspondientes a todos los {{domxref ("Cache")}} objetos rastreados por el objeto {{domxref ("CacheStorage")}} en el orden en que fueron creados. Use este método para iterar sobre una lista de todos los objetos {{domxref ("Cache")}}.
```

File: files/es/web/api/canvasrenderingcontext2d/beginpath/index.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  - API
+  - Canvas
+  - CanvasRenderingContext2D
+  - Referencia
+  - metodo
+translation_of: Web/API/CanvasRenderingContext2D/beginPath
+---
+{{APIRef}}
+
+El método **`CanvasRenderingContext2D`\*\***`.beginPath()`\*\*del API Canvas 2D comienza una nueva ruta vaciando la lista de sub-rutas. Invoca este método cuando quieras crear una nueva ruta.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/beginpath/index.md:14]:
```suggestion
El método **`CanvasRenderingContext2D.beginPath()`** del API Canvas 2D comienza una nueva ruta vaciando la lista de sub-rutas. Invoca este método cuando quieras crear una nueva ruta.
```

File: files/es/web/api/canvasrenderingcontext2d/clearrect/index.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
+---
+title: CanvasRenderingContext2D.clearRect()
+slug: Web/API/CanvasRenderingContext2D/clearRect
+translation_of: Web/API/CanvasRenderingContext2D/clearRect
+---
+{{APIRef}}El método **`CanvasRenderingContext2D`\*\***`.clearRect()`\*\*`del API Canvas 2D convierte todos los pixeles en el rectangulo definido por el punto de inicio (x, y) y tamaño`_(width, height)_ a negro transparente, borrando cualquier contenido dibujado anteriormente.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/clearrect/index.md:6]:
```suggestion
{{APIRef}}

El método **`CanvasRenderingContext2D.clearRect()`** del API Canvas 2D convierte todos los pixeles en el rectangulo definido por el punto de inicio (x, y) y tamaño _(width, height)_ a negro transparente, borrando cualquier contenido dibujado anteriormente.
```

File: files/es/web/api/canvasrenderingcontext2d/clearrect/index.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+- y
+  - : El eje **y** de la coordenada para el punto de inicio del rectangulo.
+- width
+  - : El ancho del rectangulo.
+- heigth
+  - : el alto del rectangulo.
+
+## Notas de uso
+
+Un problema común con **`clearRect`**es que puede parecer que no funciona cuando no se usan las [trayectorias de dibujo](/es/docs/Web/API/Canvas_API/Tutorial/Drawing_shapes#Drawing_paths) ([paths](/es/docs/Web/API/Canvas_API/Tutorial/Drawing_shapes#Drawing_paths)) de forma adecuada. No olvide llamar {{domxref("CanvasRenderingContext2D.beginPath", "beginPath()")}} antes de comenzar a dibujar el nuevo cuadro después de llamar **`clearRect`**`.`
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/clearrect/index.md:29]:
```suggestion
Un problema común con **`clearRect`** es que puede parecer que no funciona cuando no se usan las [trayectorias de dibujo](/es/docs/Web/API/Canvas_API/Tutorial/Drawing_shapes#Drawing_paths) ([paths](/es/docs/Web/API/Canvas_API/Tutorial/Drawing_shapes#Drawing_paths)) de forma adecuada. No olvide llamar {{domxref("CanvasRenderingContext2D.beginPath", "beginPath()")}} antes de comenzar a dibujar el nuevo cuadro después de llamar **`clearRect`**.
```

File: files/es/web/api/canvasrenderingcontext2d/clearrect/index.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+});
+
+edit.addEventListener("click", function() {
+  textarea.focus();
+})
+
+textarea.addEventListener("input", drawCanvas);
+window.addEventListener("load", drawCanvas);
+```
+
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/clearrect/index.md:113]:
```suggestion
```

File: files/es/web/api/canvasrenderingcontext2d/clearrect/index.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+ctx.clearRect(10, 10, 100, 100);
+
+// clear the whole canvas
+// ctx.clearRect(0, 0, canvas.width, canvas.height);
+```
+
+Edite el código de abajo y vea sus cambios actualizados en vivo en el canvas:
+
+{{ EmbedLiveSample('Playable_code', 700, 400) }}
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/clearrect/index.md:63]:
````suggestion
#### JavaScript

```js
var canvas = document.getElementById('canvas');
var ctx = canvas.getContext('2d');

ctx.beginPath(); ctx.moveTo(20, 20);
ctx.lineTo(200, 20);
ctx.lineTo(120, 120);
ctx.closePath(); // draws last line of the triangle
ctx.stroke();

ctx.clearRect(10, 10, 100, 100);

// clear the whole canvas
// ctx.clearRect(0, 0, canvas.width, canvas.height);
```

Edite el código de abajo y vea sus cambios actualizados en vivo en el canvas:

```html hidden
<canvas id="canvas" width="400" height="200" class="playable-canvas"></canvas>
<div class="playable-buttons">
  <input id="edit" type="button" value="Edit" />
  <input id="reset" type="button" value="Reset" />
</div>
<textarea id="code" class="playable-code" style="height:140px;">
ctx.beginPath();
ctx.moveTo(20,20);
ctx.lineTo(200,20);
ctx.lineTo(120,120);
ctx.closePath(); // draws last line of the triangle
ctx.stroke();

ctx.clearRect(10, 10, 100, 100);</textarea>
```

```js hidden
var canvas = document.getElementById("canvas");
var ctx = canvas.getContext("2d");
var textarea = document.getElementById("code");
var reset = document.getElementById("reset");
var edit = document.getElementById("edit");
var code = textarea.value;

function drawCanvas() {
  ctx.clearRect(0, 0, canvas.width, canvas.height);
  eval(textarea.value);
}

reset.addEventListener("click", function() {
  textarea.value = code;
  drawCanvas();
});

edit.addEventListener("click", function() {
  textarea.focus();
})

textarea.addEventListener("input", drawCanvas);
window.addEventListener("load", drawCanvas);
```

{{ EmbedLiveSample('Playable_code', 700, 400) }}
````

File: files/es/web/api/canvasrenderingcontext2d/fillrect/index.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
+---
+title: CanvasRenderingContext2D.fillRect()
+slug: Web/API/CanvasRenderingContext2D/fillRect
+translation_of: Web/API/CanvasRenderingContext2D/fillRect
+---
+{{APIRef}}
+
+El método **`CanvasRenderingContext2D`\*\***`.fillRect()`\*\* de la API Canvas 2D dibuja un rectángulo relleno en la posición ( x, y ). El tamaño del rectángulo se determina por width (anchura) y height (altura). El estilo de relleno se determina por el atributo `fillStyle`.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/fillrect/index.md:8]:
```suggestion
El método **`CanvasRenderingContext2D.fillRect()`** de la API Canvas 2D dibuja un rectángulo relleno en la posición ( x, y ). El tamaño del rectángulo se determina por width (anchura) y height (altura). El estilo de relleno se determina por el atributo `fillStyle`.
```

File: files/es/web/api/canvasrenderingcontext2d/linecap/index.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
+---
+title: CanvasRenderingContext2D.lineCap
+slug: Web/API/CanvasRenderingContext2D/lineCap
+translation_of: Web/API/CanvasRenderingContext2D/lineCap
+---
+{{APIRef}}
+
+La propiedad **`CanvasRenderingContext2D`\*\***`.lineCap`\*\* del API Canvas 2D determina la forma usada para dibujar los puntos finales de las líneas.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/linecap/index.md:8]:
```suggestion
La propiedad **`CanvasRenderingContext2D.lineCap`** del API Canvas 2D determina la forma usada para dibujar los puntos finales de las líneas.
```

File: files/es/web/api/canvasrenderingcontext2d/rotate/index.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
+---
+title: CanvasRenderingContext2D.rotate()
+slug: Web/API/CanvasRenderingContext2D/rotate
+tags:
+  - metodo
+translation_of: Web/API/CanvasRenderingContext2D/rotate
+---
+{{APIRef}}
+
+El método **`CanvasRenderingContext2D`\*\***`.rotate()`\*\* de la API Canvas 2D añade una rotación a la matriz de transformación.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/rotate/index.md:10]:
```suggestion
El método **`CanvasRenderingContext2D.rotate()`** de la API Canvas 2D añade una rotación a la matriz de transformación.
```

File: files/es/web/api/canvasrenderingcontext2d/save/index.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  - API
+  - Canvas
+  - CanvasRenderingContext2D
+  - Referencia
+  - metodo
+translation_of: Web/API/CanvasRenderingContext2D/save
+---
+{{APIRef}}
+
+El método **`CanvasRenderingContext2D`\*\***`.save()`\*\* del API Canvas 2D guarda el estado completo del canvas añadiendo el estado actual a una pila.
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/save/index.md:14]:
```suggestion
El método **`CanvasRenderingContext2D.save()`** del API Canvas 2D guarda el estado completo del canvas añadiendo el estado actual a una pila.
```

File: files/es/web/api/canvasrenderingcontext2d/index.md [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+  - : Directionality. Possible values: `ltr, rtl`, `inherit` (default).
+
+## Estilo del relleno y de los bordes
+
+Fill styling es utilizado para estilizar los colores del relleno y los bordes de las formas.
+
+- {{domxref("CanvasRenderingContext2D.fillStyle")}}
+  - : Utilizado para dar color al relleno de las formas. Color por defecto `#000` (negro).
+- {{domxref("CanvasRenderingContext2D.strokeStyle")}}
+  - : Utilizado para dar color al borde de las formas. Color por defecto `#000` (negro)..
```
Comment (Graywolf9) [inline files/es/web/api/canvasrenderingcontext2d/index.md:87]:
```suggestion
  - : Utilizado para dar color al borde de las formas. Color por defecto `#000` (negro).
```

File: files/es/web/api/canvas_api/tutorial/advanced_animations/index.md [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+
+## Aceleración
+
+Para convertir la moción en más auténtica, puedes jugar con la velocidad, así por ejemplo:
+
+```js
+ball.vy *= .99;
+ball.vy += .25;
+```
+
```
Comment (Graywolf9) [inline files/es/web/api/canvas_api/tutorial/advanced_animations/index.md:173]:
```suggestion
```

File: files/es/web/api/canvas_api/tutorial/advanced_animations/index.md [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+  raf = window.requestAnimationFrame(draw);
+});
+
+canvas.addEventListener('mouseout', function(e) {
+  window.cancelAnimationFrame(raf);
+});
+
+ball.draw();
+```
+
```
Comment (Graywolf9) [inline files/es/web/api/canvas_api/tutorial/advanced_animations/index.md:231]:
````suggestion

```js
ball.vy *= .99;
ball.vy += .25;
```

````

File: files/es/web/api/canvas_api/tutorial/advanced_animations/index.md [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+## Efecto de rezagar
+
+Hasta este punto hemos limpiado los fotogramas anteriores con el método {{domxref("CanvasRenderingContext2D.clearRect", "clearRect")}}. Si reemplazas este método con un semi-transparente {{domxref("CanvasRenderingContext2D.fillRect", "fillRect")}}, puedes facilmente crear un efecto de rezagar.
+
+```js
+ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
+ctx.fillRect(0, 0, canvas.width, canvas.height);
+```
+
```
Comment (Graywolf9) [inline files/es/web/api/canvas_api/tutorial/advanced_animations/index.md:242]:
```suggestion
```

File: files/es/web/api/canvas_api/tutorial/advanced_animations/index.md [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+  raf = window.requestAnimationFrame(draw);
+});
+
+canvas.addEventListener('mouseout', function(e) {
+  window.cancelAnimationFrame(raf);
+});
+
+ball.draw();
+```
+
```
Comment (Graywolf9) [inline files/es/web/api/canvas_api/tutorial/advanced_animations/index.md:299]:
````suggestion

```js
ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
ctx.fillRect(0, 0, canvas.width, canvas.height);
```

````

File: files/es/web/api/canvas_api/tutorial/compositing/index.md [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+
+Se utiliza `clip()` en lugar de `closePath()` para cerrar el trazado y volverlo uno de recorte, en vez de marcar (stroke) o rellenar el trazado.
+
+Por edfecto el elemento {{HTMLElement("canvas")}} tiene un trazado de recorte que es de extacamente el mismo tamaño del propio lienzo. En otras palabras, la máscara de recorte (clipping) no se da.
+
+### Un ejemplo de `clip`
+
+En este ejemplo, utilizamos un trazado de recorte de forma circular para restringir el dibujo de un conjunto de estrellas aleatorias dentro de una región particular del lienzo.
+
+```
```
Comment (Graywolf9) [inline files/es/web/api/canvas_api/tutorial/compositing/index.md:51]:
````suggestion
```js
````

File: files/es/web/api/console/dir/index.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+## Sintaxis
+
+```
+console.dir(objeto);
+```
+
+## Parámetros
+
+- `objeto`
+  - : `Un objeto JavaScript cuyas propiedades deben ser mostradas como resultado.`
```
Comment (Graywolf9) [inline files/es/web/api/console/dir/index.md:25]:
```suggestion
  - : Un objeto JavaScript cuyas propiedades deben ser mostradas como resultado.
```

File: files/es/web/api/console/timeend/index.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+
+## Sintaxis
+
+```
+console.timeEnd(etiqueta);
+```
+
+### Parámetros
+
+- etiqueta
```
Comment (Graywolf9) [inline files/es/web/api/console/timeend/index.md:22]:
```suggestion
- `etiqueta`
```

File: files/es/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 21]
```diff
@@ -0,0 +1,374 @@
...
+
+## Ejemplo: Crear una tabla HTML dinámicamente (`Ejemplo1.html`)
+
+### Contenido HTML
+
+```html
+<input type="button" value="Genera una tabla" onclick="genera_tabla()">
+```
+
+### JavaScript Content
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:23]:
````suggestion
Contenido HTML

```html
<input type="button" value="Genera una tabla" onclick="genera_tabla()">
```

JavaScript Content
````

File: files/es/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md [hunk 22]
```diff
@@ -0,0 +1,374 @@
...
+
+```html
+<body>
+  <input type="button" value="Set paragraph background color" onclick="set_background()">
+  <p>hi</p>
+  <p>hello</p>
+</body>
+```
+
+### JavaScript Content
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/traversing_an_html_table_with_javascript_and_dom_interfaces/index.md:141]:
````suggestion
HTML Content

```html
<body>
  <input type="button" value="Set paragraph background color" onclick="set_background()">
  <p>hi</p>
  <p>hello</p>
</body>
```

JavaScript Content
````

File: files/es/web/api/document_object_model/whitespace/index.md [hunk 23]
```diff
@@ -0,0 +1,485 @@
...
+
+Este ejemplo:
+
+```html
+<h1>◦◦◦¡Hola◦⏎
+⇥⇥⇥⇥<span>◦mundo!</span>⇥◦◦</h1>
+```
+
+se representa en el navegador así:
+
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/whitespace/index.md:86]:
```suggestion

#### Ejemplo

```

File: files/es/web/api/document_object_model/whitespace/index.md [hunk 24]
```diff
@@ -0,0 +1,485 @@
...
+```html hidden
+<h1>   ¡Hola
+    <span> mundo!</span>   </h1>
+```
+
+{{EmbedLiveSample('Hidden_example')}}
+
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/whitespace/index.md:93]:
```suggestion

#### Elemento `h1`

```

File: files/es/web/api/document_object_model/whitespace/index.md [hunk 25]
```diff
@@ -0,0 +1,485 @@
...
+⇥<div>◦◦¡Hola◦◦</div>⏎
+⏎
+◦◦◦<div>◦◦mundo!◦◦</div>◦◦⏎
+</body>
+```
+
+Tenemos 3 nodos de texto que contienen solo espacios en blanco, uno antes del primer `<div>`, uno entre los 2 `<div>`s y uno después del segundo `<div>`.
+
+Esto se renderiza así:
+
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/whitespace/index.md:157]:
```suggestion

#### Ejemplo

```

File: files/es/web/api/document_object_model/whitespace/index.md [hunk 26]
```diff
@@ -0,0 +1,485 @@
...
+```html hidden
+<body>
+  <div>  ¡Hola  </div>
+
+   <div>  mundo!   </div>
+</body>
+```
+
+{{EmbedLiveSample('Hidden_example_2')}}
+
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/whitespace/index.md:167]:
```suggestion

#### Breve resumen

```

File: files/es/web/api/document_object_model/whitespace/index.md [hunk 27]
```diff
@@ -0,0 +1,485 @@
...
+◦◦<li></li>⏎
+
+◦◦<li></li>⏎
+
+</ul>
+```
+
+Esto se traduce de la siguiente manera:
+
```
Comment (Graywolf9) [inline files/es/web/api/document_object_model/whitespace/index.md:237]:
```suggestion

#### Ejemplo

```

File: files/es/web/api/console/assert/index.md [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+  - consola web
+translation_of: Web/API/console/assert
+---
+{{APIRef("Console API")}}{{Non-standard_header}}
+
+Aparece un mensaje de error en la consola si la afirmación es falsa. Si la afirmación es verdadera, no aparecerá nada.
+
+{{AvailableInWorkers}}
+
+> **Nota:**El método `console.assert()` se implementa de diferente manera en Node.js que el mismo método disponible en los navegadores.
```
Comment (Graywolf9) [inline files/es/web/api/console/assert/index.md:22]:
```suggestion
> **Nota:** El método `console.assert()` se implementa de diferente manera en Node.js que el mismo método disponible en los navegadores.
```

**Pull request merged**